ouroboros-consensus/test-consensus/Test/Dynamic/Network.hs

```diff
@@ -140,37 +193,70 @@ broadcastNetwork registry btime numCoreNodes pInfo initRNG numSlots = do
             , blockMatchesHeader = matchesSimpleHeader
             }
 
-      node <- nodeKernel nodeParams
-
-      forM_ (filter (/= us) nodeIds) $ \them -> do
-        let mkCommsDown :: Show bytes
-                        => (   NodeChan m (SimpleBlock c ext) (SimpleHeader c ext)
-                            -> Channel m bytes
-                           )
-                        -> Codec ps e m bytes -> NodeComms m ps e bytes
-            mkCommsDown getChan codec = NodeComms {
-                ncCodec    = codec
-              , ncWithChan = \cc -> cc $
-                  loggingChannel (TalkingToConsumer us them) $
-                    getChan (chans Map.! us Map.! them)
-              }
-            mkCommsUp :: Show bytes
-                      => (   NodeChan m (SimpleBlock c ext) (SimpleHeader c ext)
-                          -> Channel m bytes
-                         )
-                      -> Codec ps e m bytes -> NodeComms m ps e bytes
-            mkCommsUp getChan codec = NodeComms {
-                ncCodec    = codec
-              , ncWithChan = \cc -> cc $
-                  loggingChannel (TalkingToProducer us them) $
-                    getChan (chans Map.! them Map.! us)
+          ni :: NetworkInterface m NodeId (AnyProtocolMessage (SimpleHeader c ext) (SimpleBlock c ext))
```

Comment:
This is too much code, the flow is gone. Move it to some local or top-level helpers.